### DISCUSSION TITLE
Does bfe support hot-restart and drain connection feature ?

### DISCUSSION THREAD

**Discussion by zhangbo1882** [bfenetworks/bfe#914] [Q&A (问题和回答)] [answered]
hot-restart :  when you upgrade the bfe version, the traffic should not be affected.
drain connection: if one instance of the cluster need to be upgrade, how to ensure the existing traffic which should go to the instance is not affected ?

**Comment by iyangsj** [comment, marked as answer]
BFE支持优雅退出：
- BFE实例在优雅退出时，将首先停止接收新连接，然后完成活跃请求的处理后再退出
- 适用于集群模式部署（需控制批量重启并发度，避免变更风险）

参考：https://www.bfe-networks.net/zh_cn/operation/signal/
